### ISSUE TITLE
Allow user to change name

### ISSUE THREAD

**Issue by Hanal-code** [techmatters/terraso-web-client#126]
## Description
Display given name and surname fields. Allow both to be edited and changes to be persisted to the database.

**Comment by josebui** [comment]
The scope of this issue changed from the conversation in: https://tech-matters.slack.com/archives/C02K02L3Q48/p1642086051064200

The scope now is:

- Only one name will be stored, and that will be `full name`
- From our auth providers this full name will be formed like:
Google: `given name + family name`
Apple: `first name + last name`

**Comment by josebui** [comment]
Copying @DerekCaelin summary here:
To summarize our conversation, we've revised our approach to this issue. We value:

- Inclusion of people who do not use the "first name" + "last name" convention
- The ability for the user to define how they are greeted
- The option to use informal speech to greet the user

For the time being, we have decided to:

- Accept the information from SSO provider and present it back to the user as given name and family name fields
- Allow the user to edit those fields
- Only require given name as mandatory
- Explain to the user that we will be addressing them by their given name

We will test our assumption that this feels inclusive to our users. If it isn't, we can explore alternatives.
